src/oauth-tooling.ts

```diff
@@ -246,6 +245,28 @@ function getAccessToken(options: OAuthConfig, logger?: Logger): Promise<Token> {
   });
 }
 
+function getCredentials(options: OAuthConfig) {
+  let credentialsPromises: Promise<string>[];
+
+  if (isCredentialsDirConfig(options)) {
+    credentialsPromises = [getFileData(options.credentialsDir, CLIENT_JSON)];
+    // For PASSWORD_CREDENTIALS_GRANT wen need user credentials as well
+    if (options.grantType === OAuthGrantType.PASSWORD_CREDENTIALS_GRANT) {
+      credentialsPromises.push(getFileData(options.credentialsDir, USER_JSON));
+    }
+  } else {
+    const clientCredentials = extractClientCredentials(options);
+    credentialsPromises = [Promise.resolve(clientCredentials)];
+    // For PASSWORD_CREDENTIALS_GRANT wen need user credentials as well
+    if (isPasswordGrantWOCredentialsDir(options)) {
```

Comment:
Would a test `isPasswordGrant` not be enough here? We are already in the branch where no credentialsDir is given, or is that not enough information for the compiler?